File: Algorithms/946_Validate_Stack_Sequences/Solution.java
import java.util.Stack;

class Solution {

    public boolean validateStackSequences(int[] pushed, int[] popped) {
        Stack<Integer> stk = new Stack<>();
        int i = 0;
        for (int element : pushed) {
            stk.push(element);
            while (!stk.isEmpty() && stk.peek() == popped[i]) {
                stk.pop();
                ++i;
            }
        }
        return stk.isEmpty();
    }
}

File: Algorithms/946_Validate_Stack_Sequences/Solution.kt
import java.util.*

class Solution {
    fun validateStackSequences(pushed: IntArray, popped: IntArray): Boolean =
        Stack<Int>().also {
            var i = 0
            for (element in pushed) {
                it.push(element)
                while (it.isNotEmpty() && it.peek() == popped[i]) {
                    it.pop()
                    ++i
                }
            }
        }.isEmpty()
}